src/app/modules/dashboard/dashboard.component.ts

```diff
@@ -127,41 +130,47 @@ export class DashboardComponent implements OnInit, OnDestroy {
     try {
       const stats = await this._apiService.getNotificationStats().toPromise();
       const dashboardStatsV2 = stats && stats["data"] ? stats["data"] : null;
-      if(dashboardStatsV2) {
+      this.callLogsByUsers = dashboardStatsV2['callsPerUser'];
+      if (dashboardStatsV2) {
         this.notificationStats = [
-          {label: 'Email Sent', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["totalEmailCount"]},
-          {label: 'Email Viewed', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["respondedEmailCount"]},
-          {label: 'Text Sent', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["totalTextCount"]},
-          {label: 'Text Viewed', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["respondedTextCount"]},
+          { label: 'Email Sent', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["totalEmailCount"] },
+          { label: 'Email Viewed', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["respondedEmailCount"] },
+          { label: 'Text Sent', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["totalTextCount"] },
+          { label: 'Text Viewed', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["respondedTextCount"] },
         ]
-
         this.callLogs = [
-          {label: 'All Calls', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["allCandidate"]},
-          {label: 'Call Again', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callAgain"]},
-          {label: 'Missed Call', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callMissed"]},
-          {label: 'Not Reachable', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callNotConnected"]},
-          {label: 'Calls on Scheduled Time', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callScheduledSuccess"]},
-          {label: 'Call Successful', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callSuccess"]},
+          { label: 'All Calls', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["allCandidate"] },
+          { label: 'Call Again', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callAgain"] },
+          { label: 'Missed Call', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callMissed"] },
+          { label: 'Not Reachable', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callNotConnected"] },
+          { label: 'Calls on Scheduled Time', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callScheduledSuccess"] },
+          { label: 'Call Successful', date: dashboardStatsV2["month_days"], data: dashboardStatsV2["callSuccess"] },
         ]
       }
     } catch (error) {
       console.log(error);
     }
   }
-}
 
+  setCallLogFiylterType(event) {
+    if (event.value === 'byUser') this.userSelected(this.callLogsByUsers[0].label);
```

Comment:
add a condition to check whether callLogsByUsers exist or not.
add a error notification as well using snackbar